src/Analyser/Analyser.php

```diff
@@ -104,6 +109,42 @@ public function analyse(
 			$postFileCallback(1);
 		}
 
+		// verify serializability of collected data
+		$alreadyReported = [];
+		foreach ($collectedData as $collected) {
+			if (array_key_exists($collected->getCollectorType(), $alreadyReported)) {
+				continue;
+			}
+
+			$json = json_encode($collected->getData());
+			if ($json === false) {
+				$errors[] = $this->buildCollectedDataError(
+					sprintf(
+						'Data collected by %s is not JSON serializable.',
+						$collected->getCollectorType(),
+					),
+					$collected,
+				);
+				$alreadyReported[$collected->getCollectorType()] = true;
+
+				break;
+			}
+
+			$restored = json_decode($json);
+			if ($restored === $collected->getData()) {
+				continue;
+			}
+
+			$errors[] = $this->buildCollectedDataError(
+				sprintf(
+					'Data collected by %s is not JSON decodable and will not work in parallel analysis.',
```

Comment:
perhaps a better term would be JSON serializable because that's the name of the [interface](https://www.php.net/manual/en/class.jsonserializable.php) php uses to allow encoding an object as json?